go/types/list.go

```diff
@@ -162,24 +162,30 @@ func (l List) RemoveAt(idx uint64) List {
 type listIterFunc func(v Value, index uint64) (stop bool)
 
 func (l List) Iter(f listIterFunc) {
+	l.IterFrom(0, f)
+}
+
+type listIterAllFunc func(v Value, index uint64)
+
+func (l List) IterAll(f listIterAllFunc) {
 	idx := uint64(0)
 	cur := newCursorAtIndex(l.seq, idx)
 	cur.iter(func(v interface{}) bool {
-		if f(v.(Value), uint64(idx)) {
-			return true
-		}
+		f(v.(Value), uint64(idx))
 		idx++
 		return false
 	})
 }
 
-type listIterAllFunc func(v Value, index uint64)
-
-func (l List) IterAll(f listIterAllFunc) {
-	idx := uint64(0)
+func (l List) IterFrom(idx uint64, f listIterFunc) {
```

Comment:
Generally speaking I needed to add an `iterator/iteratorAt` method to JS: https://github.com/attic-labs/noms/blob/master/js/noms/src/list.js#L87 - it was for some noms client I wrote. It was related to needing to compose iterations over two lists :-)

If would at least bring the SDKs closer to parity (plus I think in the case of this CL, easier to reason about the merge code).